app/src/main/java/org/eyeseetea/malariacare/domain/entity/Question.java

```diff
@@ -47,6 +67,67 @@ public String getAnswerName() {
         return answerName;
     }
 
+    public void addParent(Question question, String optionUid1) {
+        parents.add(question);
+        Map<String, Boolean> matches = new HashMap<>();
```

Comment:
I don't understand well this code, please refactor this code to more semantic code, extract in methods if It's necessary where the names are explicative